### PR TITLE
os/board/rtl8730e: update timedstay interval for ble trigger

### PR DESCRIPTION
**Pull request by ritesh55555**
this commits updates the time unit value from nanoseconds to milliseconds in timedstay api of PM during ble onboarding trigger operation.

### CODE REVIEW

File: os/board/rtl8730e/src/component/bluetooth/example/ble_scatternet/scatternet.c [hunk 1]
```diff
@@ -294,7 +294,7 @@ static rtk_bt_evt_cb_ret_t ble_tizenrt_scatternet_gap_app_callback(uint8_t evt_c
         if (!conn_ind->err) {
 #ifdef CONFIG_PM
             /* Set PM suspend timer for 10 minutes */
-            if (pm_timedstay(PM_NORMAL, 600000000) != 0) {
+            if (pm_timedstay(PM_NORMAL, 600000) != 0) {
```
Comment (r-prabu) [inline os/board/rtl8730e/src/component/bluetooth/example/ble_scatternet/scatternet.c:297]: can we have this value derived in tinyara/pm/pm.h or config.h  & used here?

**Review by r-prabu — APPROVED** [review]

File: os/pm/pm_timedstay.c [hunk 2]
```diff
@@ -108,9 +108,13 @@ static void timer_timeout(int argc, uint32_t state, uint32_t pid)
...
  ************************************************************************/
 
-int pm_timedstay(enum pm_state_e state, unsigned int timer_interval)
+int pm_timedstay(enum pm_state_e state, unsigned int milliseconds)
 {
 	int pid = PIDHASH(getpid());
+	int timer_interval;
+
+	/* Converting from milliseconds to ticks */
+	timer_interval = ((milliseonds * CLOCKS_PER_SEC) /  1000);
```
Comment (sunghan-chang) [inline os/pm/pm_timedstay.c:117]: 1. Should include necessary header for `CLOCKS_PER_SEC`
2. Regardless of this change, this file uses some config, but no including <tinyara/config.h>
3. How about using `MSEC2TICK`?

Comment (ritesh55555) [reply os/pm/pm_timedstay.c]: 2. it was already present in pm.h header . Anyway , i have added it now
3. updated

Comment (sunghan-chang) [reply os/pm/pm_timedstay.c]: 2. It should not have any dependency with other header file. If we use no config in the pm.h, it's much difficult to know whether we can remove including config.h there or not. Let's include necessary header at the file.

**Review by sunghan-chang — DISMISSED** [review]

**Comment by sunghan-chang** [comment]
@ewoodev could you review and leave an approval?

**Comment by sunghan-chang** [comment]
@ritesh55555 Oh, sorry but because I merged Gaurav's PR, this has some conflicts.

File: os/pm/pm_timedsuspend.c [hunk 3]
```diff
@@ -108,9 +109,13 @@ static void timer_timeout(int argc, uint32_t domain, uint32_t pid)
  *
  ************************************************************************/
 
-int pm_timedsuspend(enum pm_domain_e domain, unsigned int timer_interval)
+int pm_timedsuspend(enum pm_domain_e domain, unsigned int milliseconds)
 {
 	int pid = PIDHASH(getpid());
+	int timer_interval;
+
```
Comment (ewoodev) [inline os/pm/pm_timedsuspend.c:116]: This value is not needed, and It has error in the pmvdbg

Please change as below

`	int ret = wd_start(wdog, MSEC2TICK(milliseconds), (wdentry_t)timer_timeout, 2, (uint32_t)domain, (uint32_t)pid);
	pmvdbg("PM is locked for pid %d and timer started for %d milisecond\n", pid, milliseconds);
`

**Review by ewoodev — CHANGES_REQUESTED** [review]

**Review by ewoodev — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**